internal/core/src/common/Vector.h

```diff
@@ -100,28 +116,81 @@ class ColumnVector final : public BaseVector {
     }
 
     void*
-    GetRawData() {
+    RawValueAt(size_t index, size_t size_of_element) override {
+        return reinterpret_cast<char*>(GetRawData()) + index * size_of_element;
+    }
+
+    bool
+    ValidAt(size_t index) override {
+        return valid_values_[index];
+    }
+
+    void*
+    GetRawData() const {
         return values_->Data();
     }
 
     void*
     GetValidRawData() {
-        return valid_values_->Data();
+        return valid_values_.data();
     }
 
     template <typename As>
-    const As*
+    As*
     RawAsValues() const {
-        return reinterpret_cast<const As*>(values_->Data());
+        return reinterpret_cast<As*>(values_->Data());
+    }
+
+    bool
+    IsBitmap() const {
+        return is_bitmap_;
     }
 
  private:
+    bool is_bitmap_;  // TODO: remove the field after implementing BitmapVector
     FieldDataPtr values_;
-    FieldDataPtr valid_values_;
+    TargetBitmap valid_values_;  // false means the value is null
```

Comment:
@smellthemoon  please help check this change.

Reply:
it's ok to do that, but why not also change the type of values_ to TargetBitmap?

Reply:
Currently `values_` in `ColumnVector` may be scalar type vectors or bitmap. We can change the type of values_ to TargetBitmap in `BitmapVector` in the future.